docs/reference/create-client.txt

```diff
@@ -694,6 +694,20 @@ Ruby Options
      - ``Float``
      - none
 
+   * - ``:srv_max_hosts``
+     - The maximum number of mongos connections that may be created
+       for sharded topologies. If this option is set to 0, there will
+       be no maximum number of connections. If the given URI resolves
+       to more hosts than ``:srv_max_hosts``, the client will ramdomly
+       choose an ``:srv_max_hosts`` sized subset of hosts.
```

Comment:
I would add here a note that the hosts that the driver ignores during client construction will never be used, i.e., if a deployment has 10 total mongoses and this option limits them to 5, and those 5 happen to become unavailable, the client will quit working completely even though the deployment overall still has 5 functional mongoses elsewhere.